tls/extensions/s2n_client_psk.c

```diff
@@ -238,6 +239,27 @@ int s2n_client_psk_recv(struct s2n_connection *conn, struct s2n_stuffer *extensi
         return S2N_SUCCESS;
     }
 
+    s2n_extension_type_id psk_ke_mode_id;
+    GUARD(s2n_extension_supported_iana_value_to_id(TLS_EXTENSION_PSK_KEY_EXCHANGE_MODES, &psk_ke_mode_id));
+    /* https://tools.ietf.org/html/rfc8446#section-4.2.9:
+     * If clients offer "pre_shared_key" without a "psk_key_exchange_modes" extension,
+     * servers MUST abort the handshake. We can safely do this check here because s2n_client_psk is
+     * required to be the last extension sent in the list.
+     */
```

Comment:
Nit: I would put this comment before this chunk of code, rather than in the middle of it. It looks a little odd in the middle :)